core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -123,8 +125,12 @@ private[spark] abstract class Task[T](
       Option(taskAttemptId),
       Option(attemptNumber)).setCurrentContext()
 
+    plugins.foreach(_.onTaskStart())
```

Comment:
What is the expectation in case `onTaskStart` fails - do we want to invoke succeeded/failed ?

Reply:
That's what I documented on https://github.com/apache/spark/pull/29977/files#diff-6a99ec9983962323b4e0c1899134b5d6R76-R78 -- argument that came to mind is that it's easy for a plugin dev to track some state in a thread-local and clean decide if it wants to perform the succeeded/failed action or not.

Happy to change it if we prefer not to put this burden on the plugin owner though.

Reply:
We catch Throwable on `ExecutorPluginContainer#onTaskStart` and siblings (see https://github.com/apache/spark/pull/29977/files#diff-5e4d939e9bb53b4be2c48d4eb53b885c162c729b9adc874f918f7701a352cdbbR157), so that's what I meant by "not propagate". I.e. if a plugin's `onTaskStart` throws, Spark will log, but won't fail the associated spark task.

Reply:
Perhaps reword it to say exceptions are ignored ?